core/templates/components/question-directives/modal-templates/question-editor-modal.component.ts

```diff
@@ -0,0 +1,236 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for question editor modal.
+ */
+
+import { CategorizedSkills, SelectSkillModalComponent } from 'components/skill-selector/select-skill-modal.component';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { ConfirmQuestionExitModalComponent } from './confirm-question-exit-modal.component';
+import { QuestionEditorSaveModalComponent } from './question-editor-save-modal.component';
+import { ShortSkillSummary } from 'domain/skill/short-skill-summary.model';
+import { Component, Input } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { ContextService } from 'services/context.service';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import { AlertsService } from 'services/alerts.service';
+import { ImageLocalStorageService } from 'services/image-local-storage.service';
+import { QuestionValidationService } from 'services/question-validation.service';
+import { QuestionUndoRedoService } from 'domain/editor/undo_redo/question-undo-redo.service';
+import { Question } from 'domain/question/QuestionObjectFactory';
+import { Rubric } from 'domain/skill/rubric.model';
+import { State } from 'howler';
+import { GroupedSkillSummaries } from 'pages/skill-editor-page/services/skill-editor-state.service';
+import { MisconceptionSkillMap } from 'domain/skill/MisconceptionObjectFactory';
+
+interface SkillLinkageModificationsArray {
+  id: string;
+  task: string;
+}
+
+interface ReturnModalObject {
+  skillLinkageModificationsArray: SkillLinkageModificationsArray[];
+  commitMessage: string;
+}
+
+@Component({
+  selector: 'oppia-question-editor-modal',
+  templateUrl: './question-editor-modal.component.html'
```

Comment:
This seems to be referring to an empty file. Is that expected?

Reply:
`$uibModal.open({
        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
          '/components/question-directives/modal-templates/' +
            'question-editor-modal.directive.html'),`

Earlier we are using `question-editor-modal.directive.html` file as an HTML file for the  `QuestionEditorModalController` but the question-editor-modal.directive.html file does not exist in the code base in develop branch so while migration I have made new empty file to migrate QuestionEditorModalController. (we are not using this component yet for future reference I have made this file)

Reply:
Hm, if it is deadcode, should we just delete and remove all references to it?